File: backend/src/routes/logs.ts
import { Hono } from 'hono';
import { zValidator } from '@hono/zod-validator';
import { LogsQuerySchema } from '../schemas/api.schema.js';
import { LogService } from '../services/LogService.js';

const app = new Hono();
const logService = new LogService();

// 查询服务器日志
app.get('/server', zValidator('query', LogsQuerySchema), async (c) => {
  const query = c.req.valid('query');
  
  try {
    const result = await logService.getServerLogs({
      startTime: query.startTime,
      endTime: query.endTime,
      page: query.page,
      pageSize: query.pageSize,
      type: query.type
    });

    return c.json(result);
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : String(error);
    return c.json({ error: errorMessage }, 500);
  }
});

// 查询请求日志
app.get('/request', zValidator('query', LogsQuerySchema), async (c) => {
  const query = c.req.valid('query');
  
  try {
    const result = await logService.getRequestLogs({
      startTime: query.startTime,
      endTime: query.endTime,
      page: query.page,
      pageSize: query.pageSize,
      statusCode: query.statusCode,
      module: query.module
    });

    return c.json(result);
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : String(error);
    return c.json({ error: errorMessage }, 500);
  }
});

// 查询定时任务日志
app.get('/scheduled-task', zValidator('query', LogsQuerySchema), async (c) => {
  const query = c.req.valid('query');
  
  try {
    const result = await logService.getScheduledTaskLogs({
      startTime: query.startTime,
      endTime: query.endTime,
      page: query.page,
      pageSize: query.pageSize,
      status: query.status
    });

    return c.json(result);
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : String(error);
    return c.json({ error: errorMessage }, 500);
  }
});

// 手动清理旧日志
app.post('/clean', async (c) => {
  try {
    const body = await c.req.json().catch(() => ({}));
    const days = body.days || 30;
    
    const result = await logService.cleanOldLogs(days);
    
    return c.json({
      message: `已清理 ${result.deleted} 条旧日志`,
      deleted: result.deleted
    });
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : String(error);
    return c.json({ error: errorMessage }, 500);
  }
});

export default app;

File: python-backend/app/api/logs.py
"""Log API routes: server logs, request logs, scheduled task logs, cleanup."""

from typing import Optional

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db

logs_router = APIRouter()


@logs_router.get("/server")
async def get_server_logs(
    startTime: Optional[int] = Query(None),
    endTime: Optional[int] = Query(None),
    page: int = Query(1, ge=1),
    pageSize: int = Query(20, ge=1, le=200),
    type: Optional[str] = Query(None),
    db: AsyncSession = Depends(get_db),
):
    """Query server logs."""
    from app.services.log_service import LogService

    svc = LogService()
    result = await svc.get_server_logs(
        db,
        start_time=startTime,
        end_time=endTime,
        page=page,
        page_size=pageSize,
        type=type,
    )
    return result


@logs_router.get("/request")
async def get_request_logs(
    startTime: Optional[int] = Query(None),
    endTime: Optional[int] = Query(None),
    page: int = Query(1, ge=1),
    pageSize: int = Query(20, ge=1, le=200),
    statusCode: Optional[int] = Query(None),
    module: Optional[str] = Query(None),
    db: AsyncSession = Depends(get_db),
):
    """Query request logs."""
    from app.services.log_service import LogService

    svc = LogService()
    result = await svc.get_request_logs(
        db,
        start_time=startTime,
        end_time=endTime,
        page=page,
        page_size=pageSize,
        status_code=statusCode,
        module=module,
    )
    return result


@logs_router.get("/scheduled-task")
async def get_scheduled_task_logs(
    startTime: Optional[int] = Query(None),
    endTime: Optional[int] = Query(None),
    page: int = Query(1, ge=1),
    pageSize: int = Query(20, ge=1, le=200),
    status: Optional[str] = Query(None),
    db: AsyncSession = Depends(get_db),
):
    """Query scheduled task logs."""
    from app.services.log_service import LogService

    svc = LogService()
    result = await svc.get_scheduled_task_logs(
        db,
        start_time=startTime,
        end_time=endTime,
        page=page,
        page_size=pageSize,
        status=status,
    )
    return result


@logs_router.post("/clean")
async def clean_old_logs(
    body: dict = None,
    db: AsyncSession = Depends(get_db),
):
    """Clean logs older than N days."""
    from app.services.log_service import LogService

    days = 30
    if body:
        days = body.get("days", 30)

    svc = LogService()
    deleted = await svc.clean_old_logs(db, days=days)

    return {
        "message": f"Cleaned {deleted} old log entries",
        "deleted": deleted,
    }
